### PR TITLE
Fix Localization Issues

### PR DESCRIPTION
**Pull request by donker**
During testing we discovered that the thread locale was not set correctly when switching languages. This PR addresses that issue.

### CODE REVIEW

**Comment by bdukes** [comment]
I've rebased these changes on the release branch

**Review by bdukes — APPROVED** [review]
Looks good to me, but will definitely need some good testing

File: DNN Platform/HttpModules/Membership/MembershipModule.cs [hunk 1]
```diff
@@ -268,5 +249,25 @@ private void OnPreSendRequestHeaders(object sender, EventArgs e)
                 }
             }
         }
+
+        private void Context_PreRequestHandlerExecute(object sender, EventArgs e)
```
Comment (mitchelsellers) [inline DNN Platform/HttpModules/Membership/MembershipModule.cs:253]: I assume this was moved here due to the new sequence requirements for the localization process?  Might we give this method a bit better name for the purpose, although the name is correct as to what event it responds to, the WHY this is important isn't here and with no comments I'm concerned that a few years down the road we will all forget again

**Review by mitchelsellers — CHANGES_REQUESTED** [review]

**Comment by valadas** [comment]
Strange the pipeline failed because of Dnn.ClientDependency, I gonna try to run it again to see if that is a fluke...

**Comment by valadas** [comment]
Looks like it's failing because MyGet is down...

**Comment by valadas** [comment]
Looks like MyGet is back up, firing up a new build

File: DNN Platform/HttpModules/Membership/MembershipModule.cs (hunk 1 above)
Comment (donker) [reply DNN Platform/HttpModules/Membership/MembershipModule.cs:253]: The problem I'm trying to solve is that membership (AuthenticateRequest) must run before this. PreRequestHandlerExecute is the last event before main processing begins, so I moved this code to run at that time so we know who the logged in user is at this point.

Comment (donker) [reply DNN Platform/HttpModules/Membership/MembershipModule.cs:253]: https://docs.microsoft.com/en-us/troubleshoot/aspnet/http-modules-handlers

Comment (donker) [reply DNN Platform/HttpModules/Membership/MembershipModule.cs:253]: BTW, I think the naming is not all that bad given that it mimics what we do for event handlers. The original name had the same convention.

Comment (mitchelsellers) [reply DNN Platform/HttpModules/Membership/MembershipModule.cs:253]: I guess my point here is, yes the method name is “consistent”. But this was found after hours of debugging and there is nothing here that says WHY we do this. 

Renaming the method, or adding comments to save future self from pain was my point here.

Comment (donker) [reply DNN Platform/HttpModules/Membership/MembershipModule.cs:253]: It's a good idea to comment. I'll add some.

**Comment by donker** [comment]
> 
> 
> I've rebased these changes on the release branch

I updated my local repo with your changes but now we have 1000+ changes ... not sure how to resolve

**Review by valadas — CHANGES_REQUESTED** [review]
Looks like this was rebased on develop instead of the release branch... I will try to fix that if I can...

**Comment by valadas** [comment]
Yeah, I am not sure how to fix that @donker @bdukes can we fix this one or we redo a new PR, I am here to help if I can do anyhting, but I am not sure how to fix this other than a cherry pick on a new PR...

**Comment by bdukes** [comment]
I've cherry-picked from your merge commit, @donker, and then rebased again on the release branch

**Review by valadas — APPROVED** [review]
Awesome, I downloaded this CI build and ran a few tests, looks fine to me in the context of a visitor, a regular user, an admin and a host both for WebAPI and WebForms.

Thanks guys.

**Review by mitchelsellers — APPROVED** [review]

**Pull request merged**